src/components/cc-addon-elasticsearch-options/cc-addon-elasticsearch-options.js

```diff
@@ -42,60 +44,69 @@ export class CcAddonElasticsearchOptions extends LitElement {
   }
 
   /**
-   * @param {ElasticAddonOption} addonOption
-   * @returns {AddonOptionWithMetadata}
+   * Returns the metadata for an Elasticsearch option (APM or Kibana) including title, warning message, details, and logo URL
+   *
+   * @param {ElasticAddonOption['name']} elasticOptionName
+   * @returns {{ title: string, warning: string, details: Node, logo: string }}
    */
-  _getApmOption({ enabled, flavor }) {
-    const description = html`
-      <div class="option-details">${i18n('cc-addon-elasticsearch-options.description.apm')}</div>
-      <div class="option-warning">
-        <cc-icon
-          .icon="${iconAlert}"
-          a11y-name="${i18n('cc-addon-elasticsearch-options.error.icon-a11y-name')}"
-          class="icon-warning"
-        ></cc-icon>
-        <p>
-          ${i18n('cc-addon-elasticsearch-options.warning.apm')}
-          ${flavor != null ? html` ${i18n('cc-addon-elasticsearch-options.warning.apm.details', flavor)} ` : ''}
-        </p>
-      </div>
-    `;
-
-    return {
-      title: 'APM',
-      logo: APM_LOGO_URL,
-      description,
-      enabled,
-      name: 'apm',
-    };
+  _getElasticsearchOptionData(elasticOptionName) {
+    switch (elasticOptionName) {
+      case 'apm':
+        return {
+          title: 'APM',
+          warning: i18n('cc-addon-elasticsearch-options.warning.apm'),
+          details: i18n('cc-addon-elasticsearch-options.details.apm'),
+          logo: APM_LOGO_URL,
+        };
+      case 'kibana':
+        return {
+          title: 'Kibana',
+          warning: i18n('cc-addon-elasticsearch-options.warning.kibana'),
+          details: i18n('cc-addon-elasticsearch-options.details.kibana'),
+          logo: KIBANA_LOGO_URL,
+        };
+    }
   }
 
   /**
    * @param {ElasticAddonOption} addonOption
    * @returns {AddonOptionWithMetadata}
    */
-  _getKibanaOption({ enabled, flavor }) {
+  _getElasticAddonOption(addonOption) {
+    const { title, warning, details, logo } = this._getElasticsearchOptionData(addonOption.name);
+
     const description = html`
-      <div class="option-details">${i18n('cc-addon-elasticsearch-options.description.kibana')}</div>
+      <div class="option-details">${details}</div>
       <div class="option-warning">
         <cc-icon
           .icon="${iconAlert}"
           a11y-name="${i18n('cc-addon-elasticsearch-options.error.icon-a11y-name')}"
           class="icon-warning"
         ></cc-icon>
-        <p>
-          ${i18n('cc-addon-elasticsearch-options.warning.kibana')}
-          ${flavor != null ? html` ${i18n('cc-addon-elasticsearch-options.warning.kibana.details', flavor)} ` : ''}
-        </p>
+        <div>
+          <p>${warning}</p>
+          <p>
+            <span class="options-warning">
+              ${i18n('cc-addon-elasticsearch-options.warning.flavor', addonOption.flavor)}
+            </span>
+            ${'monthlyCost' in addonOption.flavor
```

Comment:
issue: even though this property is not optional on `FlavorWithMonthlyCost`, imo we should add a `addonOption.flavor.monthlyCost != null` check. As we don't have a strict null check, if you had a `monthlyCost: null`, TS wouldn't mind and the code would crash.